src/stub/core_impl/base.rs

```diff
@@ -34,6 +34,43 @@ impl<T: Target, C: Connection> GdbStubImpl<T, C> {
         Ok(tid)
     }
 
+    pub(crate) fn get_sane_current_pid(
+        &mut self,
+        target: &mut T,
+    ) -> Result<Pid, Error<T::Error, C::Error>> {
+        match target.base_ops() {
+            BaseOps::SingleThread(_) => Ok(FAKE_PID),
+            BaseOps::MultiThread(ops) => ops.active_pid().map_err(Error::TargetError),
+        }
+    }
+
+    // res.write_str("S00")?;
+    // HandlerStatus::Handled
+
+    #[inline]
+    pub(crate) fn report_stop_reason(
```

Comment:
```suggestion
    pub(crate) fn report_reasonable_stop_reason(
```

just so it's clear that it's not a "correct" stop reason - just a "reasonable" one

Reply:
Sure. Seems... reasonable.